server.go

```diff
@@ -171,22 +175,30 @@ func WithTemplateFS(fs fs.FS) func(*Server) {
 	return func(c *Server) { c.fs = fs }
 }
 
-// WithCorsMiddleware registers a middleware to handle CORS.
-// It is not handled like other middlewares with [Use] because it applies routes that are not registered.
-// For example:
+
+// WithGlobalMiddleware adds middleware(s) that will be executed on ALL requests,
+// even those that don't match any registered routes.
+// For example, to add CORS middleware:
 //
 //	import "github.com/rs/cors"
 //
 //	s := fuego.NewServer(
-//		WithCorsMiddleware(cors.New(cors.Options{
+//		WithGlobalMiddleware(cors.New(cors.Options{
 //			AllowedOrigins:   []string{"*"},
 //			AllowedMethods:   []string{"GET", "POST", "PUT", "DELETE", "OPTIONS"},
 //			AllowedHeaders:   []string{"*"},
 //			AllowCredentials: true,
-//		}).Handler)
+//		}).Handler),
 //	)
+func WithGlobalMiddleware(middlewares ...func(http.Handler) http.Handler) func(*Server) {
+	return func(c *Server) {
+		c.globalMiddlewares = append(c.globalMiddlewares, middlewares...)
+	}
+}
+
+// WithCorsMiddleware is deprecated: Use WithGlobalMiddleware instead.
```

Comment:
Please use the same syntax than all other deprecated notices, the one understood by Go editors and godoc : skip line then write `Deprecated: xxx`.

Also, please reference `[WithGlobalMiddlewares]` with brackets so it can be recognized by other Go tools like Godoc :)